aws_lambda_powertools/utilities/data_classes/vpc_lattice.py

```diff
@@ -140,3 +117,137 @@ def get_header_value(
+
+    @property
+    def query_string_parameters(self) -> Dict[str, str]:
+        """The request query string parameters."""
+        return self["query_string_parameters"]
+
+
+class vpcLatticeEventV2Identity(DictWrapper):
+    @property
+    def source_vpc_arn(self) -> Optional[str]:
+        """The VPC Lattice v2 Event requestContext Identity sourceVpcArn"""
+        return self.get("sourceVpcArn")
+
+    @property
+    def get_type(self) -> Optional[str]:
+        """The VPC Lattice v2 Event requestContext Identity type"""
+        return self.get("type")
+
+    @property
+    def principal(self) -> Optional[str]:
+        """The VPC Lattice v2 Event requestContext principal"""
+        return self.get("principal")
+
+    @property
+    def principal_org_id(self) -> Optional[str]:
+        """The VPC Lattice v2 Event requestContext principalOrgID"""
+        return self.get("principalOrgID")
+
+    @property
+    def session_name(self) -> Optional[str]:
+        """The VPC Lattice v2 Event requestContext sessionName"""
+        return self.get("sessionName")
+
+    @property
+    def x509_subject_cn(self) -> Optional[str]:
+        """The VPC Lattice v2 Event requestContext X509SubjectCn"""
+        return self.get("X509SubjectCn")
+
+    @property
+    def x509_issuer_ou(self) -> Optional[str]:
+        """The VPC Lattice v2 Event requestContext X509IssuerOu"""
+        return self.get("X509IssuerOu")
+
+    @property
+    def x509_san_dns(self) -> Optional[str]:
+        """The VPC Lattice v2 Event requestContext X509SanDns"""
+        return self.get("x509SanDns")
+
+    @property
+    def x509_san_uri(self) -> Optional[str]:
+        """The VPC Lattice v2 Event requestContext X509SanUri"""
+        return self.get("X509SanUri")
+
+    @property
+    def x509_san_name_cn(self) -> Optional[str]:
+        """The VPC Lattice v2 Event requestContext X509SanNameCn"""
+        return self.get("X509SanNameCn")
+
+
+class vpcLatticeEventV2RequestContext(DictWrapper):
+    @property
+    def service_network_arn(self) -> str:
+        """The VPC Lattice v2 Event requestContext serviceNetworkArn"""
+        return self["serviceNetworkArn"]
+
+    @property
+    def service_arn(self) -> str:
+        """The VPC Lattice v2 Event requestContext serviceArn"""
+        return self["serviceArn"]
+
+    @property
+    def target_group_arn(self) -> str:
+        """The VPC Lattice v2 Event requestContext targetGroupArn"""
+        return self["targetGroupArn"]
+
+    @property
+    def identity(self) -> vpcLatticeEventV2Identity:
+        """The VPC Lattice v2 Event requestContext identity"""
+        return vpcLatticeEventV2Identity(self["identity"])
+
+    @property
+    def region(self) -> str:
+        """The VPC Lattice v2 Event requestContext serviceNetworkArn"""
+        return self["region"]
+
+    @property
+    def time_epoch(self) -> float:
+        """The VPC Lattice v2 Event requestContext timeEpoch"""
+        return self["timeEpoch"]
```

Comment:
Are we converting it? IIRC you showed me the sample event and they send it as a str 🤦‍♂️

Reply:
You're right, I made a mistake.